Tests/LibRegex/TestRegex.cpp

```diff
@@ -1330,6 +1330,29 @@ TEST_CASE(optimizer_repeat_offset)
     }
 }
 
+TEST_CASE(quantified_alternation_capture_groups)
+{
+    {
+        // Ensure that (a|a?)+ captures the last meaningful match, not empty string
+        Regex<ECMA262> re("^(a|a?)+$");
```

Comment:
🤔 this seems very weird to me, if `a?` matched last then that should be the contents of the group, is this specified somewhere?
it does seem like other engines do this as well, so I'm okay with merging this anyway.

Reply:
Maybe paste a link to that as a note in the code lol

Reply:
Ooh, I would never notice that myself in a million years. I'll add a note now